src/SDKs/Search/DataPlane/Search.Tests/Tests/SkillsetsTests.cs

```diff
@@ -0,0 +1,610 @@
+
+                Skillset skillset = CreateTestOcrSkillset(1);
+
+                AzureOperationResponse<Skillset> response =
+                    searchClient.Skillsets.CreateOrUpdateWithHttpMessagesAsync(skillset.Name, skillset).Result;
+                Assert.Equal(HttpStatusCode.Created, response.Response.StatusCode);
+
+                skillset = CreateTestOcrSkillset(2, skillset.Name);
+                response =
+                    searchClient.Skillsets.CreateOrUpdateWithHttpMessagesAsync(skillset.Name, skillset).Result;
+                Assert.Equal(HttpStatusCode.OK, response.Response.StatusCode);
+            });
+        }
+
+        [Fact]
+        public void GetOcrSkillsetReturnsCorrectDefinition()
+        {
+            Run(() =>
+            {
+                SearchServiceClient searchClient = Data.GetSearchServiceClient();
+                CreateAndGetSkillset(searchClient, CreateTestOcrSkillset(1));
+            });
+        }
+
+        [Fact]
+        public void GetOcrSkillsetWithShouldDetectOrientationReturnsCorrectDefinition()
+        {
+            Run(() =>
+            {
+                SearchServiceClient searchClient = Data.GetSearchServiceClient();
+                CreateAndGetSkillset(searchClient, CreateTestOcrSkillset(repeat: 1, shouldDetectOrientation: true));
+            });
+        }
+
+        [Fact]
+        public void DeleteSkillsetIsIdempotent()
+        {
+            Run(() =>
+            {
+                SearchServiceClient searchClient = Data.GetSearchServiceClient();
+
+                Skillset skillset = CreateTestOcrSkillset(1);
+
+                // Try delete before the Skillset even exists.
+                AzureOperationResponse deleteResponse =
+                    searchClient.Skillsets.DeleteWithHttpMessagesAsync(skillset.Name).Result;
+                Assert.Equal(HttpStatusCode.NotFound, deleteResponse.Response.StatusCode);
+
+                searchClient.Skillsets.Create(skillset);
+
+                // Now delete twice.
+                deleteResponse = searchClient.Skillsets.DeleteWithHttpMessagesAsync(skillset.Name).Result;
+                Assert.Equal(HttpStatusCode.NoContent, deleteResponse.Response.StatusCode);
+
+                deleteResponse = searchClient.Skillsets.DeleteWithHttpMessagesAsync(skillset.Name).Result;
+                Assert.Equal(HttpStatusCode.NotFound, deleteResponse.Response.StatusCode);
+            });
+        }
+
+        private void CreateAndGetSkillset(SearchServiceClient searchClient, Skillset expectedSkillset)
+        {
+            try
+            {
+                searchClient.Skillsets.Create(expectedSkillset);
+                Skillset actualSkillset = searchClient.Skillsets.Get(expectedSkillset.Name);
+                AssertSkillsetEqual(expectedSkillset, actualSkillset);
+            }
+            finally
+            {
+                searchClient.Skillsets.Delete(expectedSkillset.Name);
```

Comment:
You actually don't need to clean up if the test fails. Every test gets its own search service, which gets cleaned up when its parent resource group is deleted by the test framework. This is why it takes so long to run tests in Record mode. :)

Reply:
Remove the cleanup. Thanks.